src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ArrayOfEmbeddedPointersNode.cs

```diff
@@ -102,5 +117,22 @@ public string MangledName
                 }
             }
         }
+        
+        private class EmbeddedPointerIndirectionWithSymbolAndOnMarkedCallbackNode : EmbeddedPointerIndirectionWithSymbolNode
```

Comment:
I put it in its own class deliberately to keep it isolated since I believe it will be rare to need the callback. Most things just use a reloc pointer to nodes they reference instead of plumbing through array indices.